### DISCUSSION TITLE
Touch Working with Generic ESP32 but not Espressive32?

### DISCUSSION THREAD

**Discussion by DovahBrownies** [Bodmer/TFT_eSPI#3395] [Q&A - Touch]
Hey all!

I bought an [ST7796S](https://www.amazon.com/dp/B0CKRJ81B5?ref=ppx_yo2ov_dt_b_product_details&th=1) ([LCD Wiki](http://www.lcdwiki.com/4.0inch_SPI_Module_ST7796)) along with a [generic ESP32 from Amazon](https://www.amazon.com/gp/product/B0B82BBKCY/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&th=1), connected the components, and set up the TFT_eSPI library as such:

**User_Setup_Select.h**
```cpp
// #include <User_Setup.h>           // Default setup is root library folder
#include <./TFT_eSPI_Setups/mySetup27_ST7796_ESP32.h> 
```

**./TFT_eSPI_Setups/mySetup27_ST7796_ESP32.h** (Copied from `User_Setups\Setup27_RPi_ST7796_ESP32.h`)
```cpp
// Comments removed for readability
#define USER_SETUP_ID 27
#define ST7796_DRIVER


#define TFT_MISO 19
#define TFT_MOSI 23
#define TFT_SCLK 18
#define TFT_CS   15   // Chip select control pin
#define TFT_DC    2   // Data Command control pin
#define TFT_RST   4   // Reset pin (could connect to RST pin)

#define TOUCH_CS 22     // Chip select pin (T_CS) of touch screen


#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define LOAD_FONT7  // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:.
#define LOAD_FONT8  // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
//#define LOAD_FONT8N // Font 8. Alternative to Font 8 above, slightly narrower, so 3 digits fit a 160 pixel TFT
#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts


#define SMOOTH_FONT


#define SPI_FREQUENCY  20000000


#define SPI_TOUCH_FREQUENCY  2500000
```

**Everything worked flawlessly**; Screen was rendering graphics and the touch was working.
I switched over to an [Espressif ESP32-S3-DevKitC-1-N8R8](https://www.amazon.com/gp/product/B09MHP42LY/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1) ([Pinout](https://docs.espressif.com/projects/esp-idf/en/stable/esp32s3/hw-reference/esp32s3/user-guide-devkitc-1.html#pin-layout)) and changed the pin definitions in the above header file:
```cpp
#define TFT_MISO 13
#define TFT_MOSI 11
#define TFT_SCLK 12
#define TFT_CS   10   // Chip select control pin
#define TFT_DC    7   // Data Command control pin
#define TFT_RST   6   // Reset pin (could connect to RST pin)

#define TOUCH_CS 5     // Chip select pin (T_CS) of touch screen
```

Graphics render just fine but I can't get the touch to work at all! These are my connections:

![image](https://github.com/Bodmer/TFT_eSPI/assets/35106956/58d593ab-cb1a-4291-8569-3da6408494be)

I've tried the `Touch_Controller_Demo` and `Test_Touch_Controller` demos to test on both MCUs.

Why is it working with the generic ESP but not the official Espressif one?
**Any help would be super appreciated! Thank you!**

**Comment by diephauthan** [comment]
i also had the same issue, im using esp32 s3 like you, the screen worked but the touch didnt work

**Reply by DovahBrownies** [reply]
Can you link the screen you're using?

**Reply by diephauthan** [reply]
I used an ILI9341 2.8 touch screen

https://www.amazon.com/HiLetgo-240X320-Resolution-Display-ILI9341/dp/B073R7BH1B

**Reply by DovahBrownies** [reply]
> I used an ILI9341 2.8 touch screen

You need a different driver for the touch:
https://github.com/PaulStoffregen/XPT2046_Touchscreen
